src/Features/Core/Portable/Snippets/SnippetProviders/ISnippetProvider.cs

```diff
@@ -28,7 +28,7 @@ internal interface ISnippetProvider
     /// <summary>
     /// Determines if a snippet can exist at a particular location.
     /// </summary>
-    ValueTask<bool> IsValidSnippetLocationAsync(in SnippetContext context, CancellationToken cancellationToken);
+    bool IsValidSnippetLocation(SnippetContext context, CancellationToken cancellationToken);
```

Comment:
💭 I was hesitant to make this change since this may be a public API in the future.

Reply:
As I said, by exposing `SemanticModel` in `SnippetContext` we eliminate the need for async for a potential user even if semantic snippets become public API. When you have it, all further symbol checking and stuff are normal syncronus code